### PR TITLE
Fix superseded PR check rollups

### LINKED ISSUE #1256: PR attention rollup counts superseded check runs as failures
## Source

Discord report by Mockafjongen: https://discord.com/channels/1522478522332090528/1535636404497809459/1536012411331547237

The report uses bb 0.36.0 on macOS.

## Problem

The pull request rollup returns `attention: "checks_failed"` when GitHub shows no failing checks.

A check named `conventional title` has two runs:

- The earlier run has the `CANCELLED` result.
- The later run has the `SUCCESS` result.

GitHub keeps the latest result for the check name. bb keeps both runs and counts the cancelled run as a failure.

The route returns these incorrect values:

```json
{
  "checks": {
    "state": "failing",
    "totalCount": 11,
    "passedCount": 9,
    "failedCount": 1,
    "pendingCount": 1
  },
  "attention": "checks_failed"
}
```

## Expected result

bb collapses superseded runs by check name. The latest check run controls the rollup.

The rollup returns `checks.state: "passing"` and `failedCount: 0`. It returns `attention: "blocked"` for the reported merge and review state.

### PR DESCRIPTION
**Pull request by ymichael**
## Summary

- preserve GitHub check timestamps through the host-daemon contract
- collapse check runs by name using the newest timestamp before calculating the PR rollup
- bump the host-daemon protocol for the wire-contract change
- cover non-chronological GitHub rollup ordering with a regression test

## Testing

- pnpm exec turbo run test --filter=@bb/host-workspace --filter=@bb/host-daemon-contract --filter=@bb/server --force
- pnpm exec turbo run typecheck --filter=@bb/domain --filter=@bb/host-workspace --filter=@bb/host-daemon-contract --filter=@bb/host-daemon --filter=@bb/server
- live sanity check against PR #1271: 19 raw runs collapse to 9 passing checks with 0 failures
- git diff --check

Closes #1256

### CODE REVIEW

**Comment by SawyerHood** [comment]
🚨 **SLOP COP** 🚨 · `review`

I am the SlopCop. I am reviewing this change now.

I will check security, code quality, performance, architecture, tests, and the end-to-end behavior.

File: apps/server/src/services/environments/pull-request.ts [hunk 1]
```diff
@@ -11,15 +11,37 @@ import type {
   ThreadPullRequestReviewState,
 } from "@bb/domain";
 
+function isSameOrLaterCheck(
+  candidate: GitHostPullRequestCheck,
+  current: GitHostPullRequestCheck,
+): boolean {
+  if (candidate.startedAt === null) {
```
Comment (SawyerHood) [inline apps/server/src/services/environments/pull-request.ts:18]: 🚨 `slopcop/review` — A queued rerun can stay hidden behind an older completed run.

GitHub defines `CheckRun.startedAt` as nullable. This branch makes a null candidate lose to every dated run. The `gh` JSON can also represent a missing time as year 0001, which has the same result. I reproduced an older successful run plus a newer queued run. The rollup returned `passing` with `pendingCount: 0`. Equal timestamps also depend on the input order.

Please preserve a creation time or a stable run identifier. Add tests for null, zero, and equal timestamps in both input orders.

File: apps/server/src/services/environments/pull-request.ts [hunk 2]
```diff
@@ -11,15 +11,37 @@ import type {
...
+  }
+  return Date.parse(candidate.startedAt) >= Date.parse(current.startedAt);
+}
+
 function assembleThreadPullRequestChecks(
   rawChecks: readonly GitHostPullRequestCheck[],
 ): ThreadPullRequestChecks {
+  const latestChecksByName = new Map<string, GitHostPullRequestCheck>();
+  for (const check of rawChecks) {
+    const current = latestChecksByName.get(check.name);
```
Comment (SawyerHood) [inline apps/server/src/services/environments/pull-request.ts:32]: 🚨 `slopcop/review` — Name-only grouping can combine independent checks.

A visible check name is not a stable producer identity. Separate workflows or check producers can use the same name. A later success can then replace an independent failure. The host currently removes the check kind and producer identity before this map receives the data.

Please preserve a stable workflow or producer identity. Group with that identity and the check name. Keep ambiguous checks separate when the identity is absent.

**Review by SawyerHood — COMMENTED** [review]
🚨 **SLOP COP** 🚨 · `review`

### ELI5

GitHub can send several attempts for one check. This change keeps the newest attempt before it reports the result.

Two edge cases can still hide a pending or failed check. The app can then show a green result when it should not.

### Findings

- **High:** A queued rerun can lose to an older completed run. GitHub permits a null `startedAt` value. The comparator always keeps the older dated run. A direct test returned `passing` with `pendingCount: 0`. Equal timestamps also depend on the array order.

- **High:** The map uses only the visible check name. Separate workflows or producers can use the same name. A later success can replace an independent failure. Preserve a stable producer identity and keep ambiguous checks separate.

- **Medium:** The GitHub plugin has a second check-rollup implementation. It still counts all superseded rows. Move the common GitHub normalization and selection into one internal module. Keep the product attention policy in the server.

I left inline comments for both high-severity findings.

### Snapshot note

The requested SHA omitted regenerated plugin SDK declarations. That snapshot failed the generated-output CI check. Commit `a53b8c5` fixes this item at the current PR head.

### Security and performance

I found no command injection, path escape, authorization bypass, or data exposure. GitHub still enforces merge authorization and branch protection.

The new map uses linear time over bounded check data. I found no material performance problem.

### Validation

- The exact SHA passed the full server, host-workspace, and host-daemon-contract test suites.
- The server suite passed 1,433 tests.
- The three changed test files passed 69 tests.
- Five affected package type checks passed.
- `git diff --check` passed.
- A direct test confirmed the null-time and equal-time defects.
- Dev-browser verified the public GitHub pull request workflow.
- The live product route reached the daemon, but that daemon did not have `gh`.

The pull request advanced during this review. The code findings remain fixed to `fdef28c83cc80b33af29864b2f9d78303ec1f547`.
